src/k8s_utils.py

```
@@ -167,3 +168,38 @@ def k8s_create_secret(
     except ApiError as ae:
         logging.error("Failed to create secret '%s' in namespace '%s': %s", name, namespace, ae)
         raise ae
+
+
+def k8s_create_cluster_ip_service(
+    kube_client: Client,
+    name: str,
+    namespace: str,
+    selector: Dict[str, str],
+    ports: List[ServicePort],
+    labels: Optional[Dict[str, Any]] = None,
+) -> None:
+    """Create a Kubernetes service.
+
+    Args:
+        kube_client (Client): The Kubernetes client used to interact with the cluster.
+        name (str): The name of the service.
+        namespace (str): The namespace of the service.
+        ports (List[ServicePort]): The ports for the service.
+        selector (Dict[str, str]): The selector for the service.
+        labels (Optional[Dict[str, Any]]): Optional labels for the service.
+
+    Raises:
+        ApiError: If the service cannot be created.
+    """
+    try:
+        kube_client.create(
+            Service(
+                apiVersion="v1",
+                kind="Service",
+                metadata=ObjectMeta(name=name, namespace=namespace, labels=labels),
+                spec=ServiceSpec(type="ClusterIP", selector=selector, ports=ports),
+            )
+        )
+    except ApiError as ae:
+        logging.error("Failed to create service '%s' in namespace '%s': %s", name, namespace, ae)
+        raise ae
```

**question**: Shouldn't we just let the `ApiError` go upwards in this case since it's unexpected error, instead of re-raising?

---

We can let it throw, but I've jut wanted to add extra logging or to specify what happened